components/tikv_util/src/macros.rs

```diff
@@ -171,6 +171,38 @@ macro_rules! try_opt_or {
     }};
 }
 
+#[cfg(target_os = "linux")]
+#[macro_export]
+macro_rules! add_thread_name {
+    ($f:expr) => {{
+        use $crate::metrics::THREAD_NAME_HASHMAP;
+        use $crate::nix::libc::pid_t;
+        use $crate::nix::unistd::gettid;
+        move || {
+            if let Some(name) = std::thread::current().name() {
```

Comment:
why can we correctly retrieve the name here, while name is truncated in other places like `top`?

Reply:
Because this name is stored in rust std::thread struct. The length restriction is in linux kernel. Thread name will be truncated after calling `pthread_setname_np`library function.